Engine/src/Engine/JobQueue.cpp

```diff
@@ -0,0 +1,54 @@
+#include "Log.h"
+#include "JobQueue.h"
+
+using namespace engine_concurrent;
+
+JobQueue* JobQueue::instance = nullptr;
+std::mutex JobQueue::instanceMutex;
+std::mutex JobQueue::queueMutex;
+
+JobQueue* JobQueue::getInstance()
+{
+	std::unique_lock lock(instanceMutex);
+	if (instance == nullptr)
+	{
+		instance = new JobQueue();
+	}
+	return instance;
+}
+
+JobQueue::JobQueue() : m_jobList(new std::list<ThreadJob*>()) {}
+JobQueue::~JobQueue() { delete m_jobList; }
+
+void JobQueue::postJob(ThreadJob* job)

```

Comment:
just to be sure here I would suggest changing the return type here to bool, where the return is a success state of the job being posted